### PR TITLE
Finally implementing Grid

### PR DESCRIPTION
**Pull request by romaninsh**
Merge #74 first.

This PR implements a new Grid class with a bunch of features:

 - menu / button support on top of the grid
 - quick-search input field
 - support for checkboxes
 - support for actions (buttons)

screen shot 2017-03-30 at 11 05 35

### CODE REVIEW

**Comment by romaninsh** [comment]
I must also add that this PR fixes HTML-escaping in Table.

File: src/Button.php [hunk 1]
```diff
@@ -70,24 +70,4 @@ public function renderView()
...
-        if (is_string($url)) {
-            $this->setAttr('href', $url);
-
-            return $this;
-        }
-        $this->setAttr('href', $this->app->url($url));
-
-        return $this;
-    }
 }
```
Comment (romaninsh) [inline src/Button.php]: moved to View

**Comment by romaninsh** [comment]
Actually - docs will come in the final implementation of Grid, which I'm yet to PR.

File: src/Table.php [hunk 2]
```diff
@@ -384,9 +405,19 @@ public function getRowTemplate()
     {
         $output = [];
         foreach ($this->columns as $name => $column) {
-            $field = $this->model->getElement($name);
 
-            $output[] = $column->getCellTemplate($field);
+            // If multiple formatters are defined, use the first for the header cell
```
Comment (DarkSide666) [inline src/Table.php]: If data cell has multiple formatters, then all of them should be used here. This is not header cell here !

File: src/TableColumn/Checkbox.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    public $class = null;
+
+    /**
+     * Return action which will calculate and return array of all checkbox IDs, e.g.
+     *
+     * [3, 5, 20]
+     */
+    public function jsChecked()
+    {
+        return new \atk4\ui\jsExpression(' $('.$this->table->jsRender().").find('.checked.".$this->class."').closest('tr').map(function(){ ".
```
Comment (DarkSide666) [inline src/TableColumn/Checkbox.php]: can't we use simpler JS syntax here? something like:
```
$(...).find('.checked.xxx').closest('tr').get(0).data('id');
```

File: src/Table.php (hunk 2 above)
Comment (romaninsh) [inline src/Table.php]: I'm unclear how to implement that. I need more info in #92 and #67. The mechanism is there, so I propose we just look for bugs and various combinations then address them.

File: src/TableColumn/Checkbox.php (hunk 3 above)
Comment (romaninsh) [inline src/TableColumn/Checkbox.php]: I'll refactor those later with JSLib. Will create a separate issue.

**Pull request merged**